### DISCUSSION TITLE
How to login to a browser extension?

### DISCUSSION THREAD

**Discussion by hyperknot** [zitadel/zitadel#5011] [Q&A]
I'm evaluating Auth solutions and one thing I cannot figure out with ZITADEL is if it's possible to authenticate a user for a browser extension?

I mean there are two ways to log in normally, either through an apps' own UI or with an external login hosted on a separate domain/subdomain. If I understand correctly, ZITADEL only supports the external login, is that right?

Now with an external login, how would you login to a browser extension? Has anyone ever done something like this with ZITADEL?

**Comment by dakshitha** [comment]
Yes, ZITADEL only supports external logins with the Authorization Code grant type (or Implicit if absolutely needed) for user agents such as browser extensions. To log in to a browser extension, redirect the user to the ZITADEL login page, where the user can enter their credentials. After the user is authenticated, the extension can then access the necessary information or resources. This can be done via a token passed by redirect once the login completed. Hope this answers your question. This [discussion](https://github.com/zitadel/zitadel/discussions/4193) might also be of interest to you.
